In [5]:
pacman::p_load("tidyverse", "ggpubr", 'car', 'broom')

In [15]:
df <- read_table2("STAT501_Lesson01/allentest.txt")
head(df)

Parsed with column specification:
cols(
  Subj = col_double(),
  ACL = col_double(),
  Vocab = col_double(),
  Abstract = col_double(),
  SDMT = col_double()
)



Subj ACL Vocab Abstract SDMT
1 1    6.0 28    36       70  
2 2    5.4 34    32       49  
3 3    4.7 19     8       28  
4 4    4.8 32    28       47  
5 5    4.9 22     4       29  
6 6    4.5 24    24       23

In [16]:
slm.voca <- lm(ACL ~ Vocab, data=df)
summary(slm.voca)


Call:
lm(formula = ACL ~ Vocab, data = df)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.4490 -0.6021 -0.1723  0.7106  1.5999 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   4.2253     0.3524  11.989   <2e-16 ***
Vocab         0.0298     0.0141   2.113   0.0383 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.7761 on 67 degrees of freedom
Multiple R-squared:  0.0625,	Adjusted R-squared:  0.04851 
F-statistic: 4.467 on 1 and 67 DF,  p-value: 0.03829


In [19]:
round(anova(slm.voca),2)

Df Sum Sq Mean Sq F value Pr(>F)
Vocab      1  2.69  2.69    4.47    0.04  
Residuals 67 40.36  0.60      NA      NA

In [25]:
round(Anova(slm.voca, type = 3),2)

Sum Sq Df F value Pr(>F)
(Intercept) 86.59   1 143.75  0.00  
Vocab        2.69   1   4.47  0.04  
Residuals   40.36  67     NA    NA

In [26]:
mlm.voca.sdmt <- lm(ACL ~ Vocab + SDMT, data=df)
summary(mlm.voca.sdmt)


Call:
lm(formula = ACL ~ Vocab + SDMT, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.55758 -0.44619 -0.01027  0.34114  1.55955 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3.845292   0.324380  11.854  < 2e-16 ***
Vocab       -0.006840   0.015045  -0.455    0.651    
SDMT         0.029795   0.006803   4.379 4.35e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.6883 on 66 degrees of freedom
Multiple R-squared:  0.2736,	Adjusted R-squared:  0.2516 
F-statistic: 12.43 on 2 and 66 DF,  p-value: 2.624e-05


In [34]:
round(anova(mlm.voca.sdmt),2)
## SSR = 2.69 + 9.09 = 11.77
## SSE = 31.27
## SST = 11.77 + 31.27 = 43.04

Df Sum Sq Mean Sq F value Pr(>F)
Vocab      1  2.69  2.69     5.68   0.02  
SDMT       1  9.09  9.09    19.18   0.00  
Residuals 66 31.27  0.47       NA     NA

In [33]:
round(Anova(mlm.voca.sdmt, type = 3),2)
## SSR(vocab | SDMT) = 0.1, SSR(SDMT | vocab) = 9.09

Sum Sq Df F value Pr(>F)
(Intercept) 66.58   1 140.52  0.00  
Vocab        0.10   1   0.21  0.65  
SDMT         9.09   1  19.18  0.00  
Residuals   31.27  66     NA    NA

In [37]:
round(anova(slm.voca, mlm.voca.sdmt),2)
## RSS = Residual Sum Sq. 40.36 - 31.27 = explained by SDMT variable. = Sum of Sq

Res.Df RSS   Df Sum of Sq F     Pr(>F)
1 67     40.36 NA   NA         NA NA    
2 66     31.27  1 9.09      19.18  0

In [39]:
round(anova(lm(ACL ~ SDMT + Vocab, data=df)),2)

Df Sum Sq Mean Sq F value Pr(>F)
SDMT       1 11.68  11.68   24.65   0.00  
Vocab      1  0.10   0.10    0.21   0.65  
Residuals 66 31.27   0.47      NA     NA

In [40]:
mlm.all <- lm(ACL ~ SDMT + Vocab + Abstract, data=df)
summary(mlm.all)


Call:
lm(formula = ACL ~ SDMT + Vocab + Abstract, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.63459 -0.46009 -0.02471  0.33624  1.52886 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  3.946347   0.338069  11.673  < 2e-16 ***
SDMT         0.027404   0.007168   3.823 0.000298 ***
Vocab       -0.017397   0.018077  -0.962 0.339428    
Abstract     0.012182   0.011585   1.051 0.296926    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.6878 on 65 degrees of freedom
Multiple R-squared:  0.2857,	Adjusted R-squared:  0.2528 
F-statistic: 8.668 on 3 and 65 DF,  p-value: 6.414e-05


In [42]:
round(anova(mlm.all),2)

Df Sum Sq Mean Sq F value Pr(>F)
SDMT       1 11.68  11.68   24.69   0.00  
Vocab      1  0.10   0.10    0.21   0.65  
Abstract   1  0.52   0.52    1.11   0.30  
Residuals 65 30.75   0.47      NA     NA

In [43]:
round(Anova(mlm.all, type=3),2)

Sum Sq Df F value Pr(>F)
(Intercept) 64.46   1 136.26  0.00  
SDMT         6.91   1  14.61  0.00  
Vocab        0.44   1   0.93  0.34  
Abstract     0.52   1   1.11  0.30  
Residuals   30.75  65     NA    NA

### Try it! Samples
#### IQ Size data

In [45]:
iq.df <- read_table2("STAT501_Lesson01/iqsize.txt")
head(iq.df)

Parsed with column specification:
cols(
  PIQ = col_double(),
  Brain = col_double(),
  Height = col_double(),
  Weight = col_double()
)



PIQ Brain  Height Weight
1 124  81.69 64.5   118   
2 150 103.84 73.3   143   
3 128  96.54 68.8   172   
4 134  95.15 65.0   147   
5 110  92.88 69.0   146   
6 131  99.13 64.5   138

In [46]:
slm.br <- lm(PIQ ~ Brain, data=iq.df)
summary(slm.br)


Call:
lm(formula = PIQ ~ Brain, data = iq.df)

Residuals:
    Min      1Q  Median      3Q     Max 
-40.077 -17.508  -2.095  17.097  41.571 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   4.6519    43.7118   0.106   0.9158  
Brain         1.1766     0.4806   2.448   0.0194 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 21.21 on 36 degrees of freedom
Multiple R-squared:  0.1427,	Adjusted R-squared:  0.1189 
F-statistic: 5.994 on 1 and 36 DF,  p-value: 0.01935


In [48]:
round(anova(slm.br),1)

Df Sum Sq  Mean Sq F value Pr(>F)
Brain      1  2697.1 2697.1   6       0    
Residuals 36 16197.5  449.9  NA      NA

#### 1. Fit the linear regression model with x1 = brain as the only predictor.  
What is the value of the error sum of squares, denoted SSE(Brain) since X1(Brain) is the only predictor in the model?  
  
SSE(Brain) = 16197.5   
SSR(Brain) = 2697.1  
SST = 2697 + 16197 = 18895  

#### 2. Now, fit the linear regression model with the predictors (in order) x1 = brain and x2 = height in the model.  


In [53]:
mlm.2 <- lm(PIQ ~ Brain + Height, data=iq.df)

In [52]:
round(anova(mlm.2),1)

Df Sum Sq  Mean Sq F value Pr(>F)
Brain      1  2697.1 2697.1  7.1      0    
Height     1  2875.6 2875.6  7.6      0    
Residuals 35 13321.8  380.6   NA     NA

What is the value of the error sum of squares, denoted SSE(Brain, Height) since x1 and x2 are the only predictors in the model?  
What is the value of the regression sum of squares, denoted SSR(Brain, Height) since x1 and x2 are the only predictors in the model?  
Confirm that the value of SSTO is unchanged from the previous question.  

1. SSE(Brain, Height) = 13321
2. SSR(Brain, Height) = 2697 + 2875 = 5573
3. SST = SSE + SSR = 13321 + 5573 = 18894. It is unchanged.

#### 3. Now, let's use the above definitions to calculate the sequential sum of squares of adding X2 to the model in which X1 is the only predictor. We denote this quantity as SSR(X2|X1). (The bar "|" is read as "given.") According to the alternative definitions:

SSR(X2|X1) is the reduction in the error sum of squares when X2 is added to the model in which X1 is the only predictor. That is, SSR(X2|X1)= SSE(X1) – SSE(X1,X2). What is the value of SSR(X2|X1) calculated this way?  

SSR(X2|X1) = SSE(X1) - SSE(X1, X2) = 16197.5 - 13321.8 = 2875  
SSR(X2|X1) = SSR(X1,X2) - SSR(X1) = 2875.6

In [57]:
## Reoder

mlm.5 <- lm(PIQ ~ Height + Brain, data=iq.df)
round(anova(mlm.5),1)

Df Sum Sq  Mean Sq F value Pr(>F)
Height     1   164.0  164.0   0.4    0.5   
Brain      1  5408.8 5408.8  14.2    0.0   
Residuals 35 13321.8  380.6    NA     NA

SSR(X2) = 164  
SSR(X1|X2) = 5409  
SSR(X1|X2) = SSE(X2) – SSE(X1, X2) = 18731 – 13322 = 5409  
SSR(X1|X2) = SSR(X1, X2) – SSR(X2) = 5573 – 164 = 5409  

In [60]:
mlm.all <- lm(PIQ ~ Brain + Height + Weight, data=iq.df)
round(anova(mlm.all),1)

Df Sum Sq  Mean Sq F value Pr(>F)
Brain      1  2697.1 2697.1  6.9      0    
Height     1  2875.6 2875.6  7.3      0    
Weight     1     0.0    0.0  0.0      1    
Residuals 34 13321.8  391.8   NA     NA

All of the sequential sums of squares we considered so far are "one-degree-of-freedom sequential sums of squares," because we have only considered the effect of adding one additional predictor variable to a model. We could, however, quantify the effect of adding two additional predictor variables to a model. For example, we might want to know the effect of adding X2 and X3 to a model that already contains X1 as a predictor. The sequential sum of squares SSR(X2,X3|X1) quantifies this effect. It is a "two-degree-of-freedom sequential sum of squares," because it quantifies the effect of adding two additional predictor variables to the model. One-degree-of-freedom sequential sums of squares are used in testing one slope parameter such as H0 : β1 =0, where as two-degree-of-freedom sequential sums of squares are used in testing two slope parameters, such as H0 : β1 = β2 = 0.

Use either of the two definitions to calculate SSR(X2,X3|X1). That is, calculate SSR(X2,X3|X1) by SSR(X1,X2,X3) – SSR(X1) or by SSE(X1) – SSE(X1,X2,X3).


SSE(X1) = 16197.5  
SSE(X1,X2,X3) = 13321.8  
SSE(X1) - SSE(X1,X2,X3) = 2875.7

SSR(X1,X2,X3) = 2697 + 2875 + 0 = 5572  
SSR(X1) = 2697  
SSR(X1,X2,X3) - SSR(X1) = 5572-2697 = 2876  

In [65]:
round(Anova(mlm.all, type=3),1)

Sum Sq  Df F value Pr(>F)
(Intercept)  1225.2  1  3.1    0.1   
Brain        5239.2  1 13.4    0.0   
Height       1934.7  1  4.9    0.0   
Weight          0.0  1  0.0    1.0   
Residuals   13321.8 34   NA     NA

Fit the linear regression model with y = PIQ and (in order) x1 = brain and x2 = height. Verify that the regression sum of squares obtained, SSR(X1,X2), is the sum of the two sequential sum of squares SSR(X1) and SSR(X2|X1). This illustrates how SSR(X1,X2) is "decomposed" into a sum of sequential sum of squares.

In [66]:
mlm.2


Call:
lm(formula = PIQ ~ Brain + Height, data = iq.df)

Coefficients:
(Intercept)        Brain       Height  
    111.276        2.061       -2.730  


In [68]:
round(anova(mlm.2))

Df Sum Sq Mean Sq F value Pr(>F)
Brain      1  2697  2697     7       0    
Height     1  2876  2876     8       0    
Residuals 35 13322   381    NA      NA

In [69]:
round(anova(slm.br))

Df Sum Sq Mean Sq F value Pr(>F)
Brain      1  2697  2697     6       0    
Residuals 36 16197   450    NA      NA

SSR(X1, X2) = SSR(X1) + SSR(X2|X1) = 2697 + 2876 = 5573  
SSR(X1, X2) = SSR(X2) + SSR(X1|X2) = 164 + 5409 = 5573  

In [71]:
tmp.mlm <- lm(PIQ ~ Height + Brain, data=iq.df)
round(anova(tmp.mlm))

Df Sum Sq Mean Sq F value Pr(>F)
Height     1   164   164     0       1    
Brain      1  5409  5409    14       0    
Residuals 35 13322   381    NA      NA

In [74]:
round(anova(slm.br, tmp.mlm)) 

## RSS residual sum of sq

Res.Df RSS   Df Sum of Sq F  Pr(>F)
1 36     16197 NA   NA      NA NA    
2 35     13322  1 2876       8  0